### PR TITLE
Document how nodes get their configuration in 3.10

### PR DESCRIPTION
**Pull request by mburke5678**
https://trello.com/c/u7dNzD5v/949-document-how-nodes-get-their-configuration-in-310

@mburke5678 I changed the link to the docs trello board instead of yours.

### CODE REVIEW

File: security/monitoring.adoc [hunk 1]
```diff
@@ -89,7 +89,7 @@ This section describes the types of operational logs produced on the cluster.
 [[security-monitoring-service-logs]]
 === Service Logs
 
-{product-title} produces logs with each *systemd* service that is running on a host:
+{product-title} produces logs for services that run on static pods in a cluster:
```
Comment (aveshagarwal) [inline security/monitoring.adoc]: Perhaps you also want to mention that apiserver, controllers and etcd static pods run in kube-system namespace.

File: scaling_performance/using_cpu_manager.adoc [hunk 2]
```diff
@@ -38,7 +38,13 @@ To set up CPU Manager:
 . Enable CPU manager support on the target node:
 +
 ----
-# cat /etc/origin/node/node-config.yaml
+oc edit configurationmaps <name> -n openshift-node
```
Comment (aveshagarwal) [inline scaling_performance/using_cpu_manager.adoc]: configurationmaps->configmap or cm

File: scaling_performance/managing_hugepages.adoc [hunk 3]
```diff
@@ -48,7 +48,13 @@ huge pages. This topic describes how.
...
 +
 ----
-# cat /etc/origin/node/node-config.yaml
+# oc edit configurationmaps <name>
+----
++
+For example:
++
+----
+oc edit confgurationmaps node-config-compute -n openshift-node
```
Comment (aveshagarwal) [inline scaling_performance/managing_hugepages.adoc]: configurationmaps->configmap(s) or cm

File: scaling_performance/managing_hugepages.adoc [hunk 4]
```diff
@@ -48,7 +48,13 @@ huge pages. This topic describes how.
 . `HugePages` must be set to `true` across the system:
 +
 ----
-# cat /etc/origin/node/node-config.yaml
+# oc edit configurationmaps <name>
```
Comment (aveshagarwal) [inline scaling_performance/managing_hugepages.adoc]: configurationmaps->configmap(s) or cm

**Comment by aveshagarwal** [comment]
@mburke5678 thanks for this PR. I just have some minor comments otherwise lgtm. 
lets also wait for @derekwaynecarr 's feedback.

**Comment by derekwaynecarr** [comment]
@mfojtik we need a corresponding document that describes the administrative changes for the control plane components.  it should describe the transition toward self hosting, describe how users can perform actions via the cluster (read api server, controller logs), and on the host directly (stop, start, restart).

there are a number of references to `systemctl <something> atomic-openshift-master-*` components that no longer are pertinent in this release.

for example:
https://github.com/openshift/openshift-docs/search?q=atomic-openshift-master&unscoped_q=atomic-openshift-master

**Comment by derekwaynecarr** [comment]
@aveshagarwal the content here is good, but can we add a section on node boostrapping?

having material that describes the flow similar to your CEE presentation would be a big help.
describe the sequence, discuss bootstrapper cluster role, service account, the role of bootstrap kubeconfig, and csr flow.

if we want to do that via a separate follow-on pr, that is fine.

**Comment by aveshagarwal** [comment]
@derekwaynecarr @mburke5678 @mfojtik so seems like this PR https://github.com/openshift/openshift-docs/pull/10762 is covering node bootstrapping process, and static pods based control plane stuff too in 3.10.

**Comment by mburke5678** [comment]
@mdshuai PTAL

File: admin_guide/manage_nodes.adoc [hunk 5]
```diff
@@ -477,13 +477,123 @@ In rare cases, a router pod may not have a host port configured. In those cases,
 it is important to follow the xref:infrastructure-nodes[recommended restart
 process] for infrastructure nodes.
 
+[[modifying-nodes]]
+== Modifying Nodes
+
+// tag::node-configmap[]
+During installation, {product-title} creates a configuration map in the *openshift-node* project for each type of node group:
```
Comment (mdshuai) [inline admin_guide/manage_nodes.adoc]: configmaps is specific resource, we'd better update "configuration map" to "configmaps"

File: admin_guide/manage_nodes.adoc [hunk 6]
```diff
@@ -477,13 +477,123 @@ In rare cases, a router pod may not have a host port configured. In those cases,
...
+During installation, {product-title} creates a configuration map in the *openshift-node* project for each type of node group: 
+
+* *node-config-master* for xref:../architecture/infrastructure_components/kubernetes_infrastructure.adoc#master[master nodes];
+* *node-config-infra* for xref:../admin_guide/manage_nodes.adoc#infrastructure-nodes[infrastructure nodes];
+* *node-config-compute* for compute nodes.
+
+In order to make configuration changes to an existing node, edit the appropriate configuration map. A _sync pod_ on each node
+watches for changes in the configuration maps. The sync pods are automatically created by the installer using _sync daemonsets_ and
+create a *_/etc/origin/node/node-config.yaml_* file on each node, where the node configuration parameters reside. When a sync pod
+detects configuration map change, it updates the *_node-config.yaml_* on all nodes in that node group.
```
Comment (mdshuai) [inline admin_guide/manage_nodes.adoc]: After update the node-config.yaml, The sync pod also will kill the `hyperkube kubelet` process to trigger node restart use the updated configure file.

**Comment by mburke5678** [comment]
@aveshagarwal Can I ask you to take one more look? I added some call-outs to the config map example. I pulled the info from other places in the docs and want to make sure they are OK in this context.
https://github.com/openshift/openshift-docs/pull/10575/files#diff-a152e3b2fc039686d4d65b7a3c719b4dR583

**Comment by aveshagarwal** [comment]
@mburke5678 lgtm.

**Comment by mburke5678** [comment]
@openshift/team-documentation PTAL. Most changes are find/replace. Some new text

File: scaling_performance/using_cpu_manager.adoc [hunk 7]
```diff
@@ -38,7 +38,13 @@ To set up CPU Manager:
 . Enable CPU manager support on the target node:
 +
 ----
-# cat /etc/origin/node/node-config.yaml
+oc edit configmap <name> -n openshift-node
```
Comment (kalexand-rh) [inline scaling_performance/using_cpu_manager.adoc]: $

File: scaling_performance/using_cpu_manager.adoc [hunk 8]
```diff
@@ -38,7 +38,13 @@ To set up CPU Manager:
...
+----
++
+For example:
++
+----
+oc edit cm node-config-compute -n openshift-node
```
Comment (kalexand-rh) [inline scaling_performance/using_cpu_manager.adoc]: $

File: security/monitoring.adoc [hunk 9]
```diff
@@ -104,9 +104,14 @@ ifdef::openshift-enterprise[]
 - atomic-openshift-node
 endif::[]
 
-These logs are intended more for debugging purposes than for security auditing. You can retrieve logs for 
-each service with the `master-logs api api`, `master-logs controllers controllers`, or `master-logs etcd etcd` commands. 
-If your cluster runs an aggregated logging stack, like an xref:../install_config/aggregate_logging.adoc#aggregated-ops[Ops cluster], cluster administrators can retrieve logs from the logging _.operations_ indexes.
+These logs are intended more for debugging purposes than for security auditing. They can be
```
Comment (kalexand-rh) [inline security/monitoring.adoc]: :crying_cat_face: 
Please make these sentences active. I don't have it left in me.

**Review by kalexand-rh — COMMENTED** [review]
Picks!

File: admin_guide/manage_nodes.adoc [hunk 10]
```diff
@@ -477,13 +485,137 @@ In rare cases, a router pod may not have a host port configured. In those cases,
...
+<2> IP address to be prepended to a pod’s *_/etc/resolv.conf_*.
+<3> Key value pairs that are passed directly to the Kubelet that match the link:https://kubernetes.io/docs/admin/kubelet/[Kubelet's command line arguments].
+<4> The path for the pod manifest file or directory. A directory must contain one or more manifest files. 
+{product-title} uses the manifest files to create pods on the node.
+<5> The pod network settings on the node
+<6> Software defined network (SDN) plug-in. Set to `redhat/openshift-ovs-subnet` for the *ovs-subnet* plug-in,
+`redhat/openshift-ovs-multitenant` for the *ovs-multitenant* plug-in, or
+`redhat/openshift-ovs-networkpolicy` for the *ovs-networkpolicy* plug-in
+<7> Certificate information for the node. 
+<8> Optional: PEM-encoded certificate bundle. If set, a valid client certificate
```
[the remark this answers is not in the document]
Comment (mburke5678) [reply admin_guide/manage_nodes.adoc]: @kalexand-rh I copied this and most of these from elsewhere in our docs.

File: admin_guide/manage_nodes.adoc [hunk 11]
```diff
@@ -495,22 +627,22 @@ kubeletArguments:
...
++
 <1> xref:../admin_guide/manage_nodes.adoc#admin-guide-max-pods-per-node[Maximum number of pods that can run on this kubelet].
 <2> Resolver configuration file used as the basis for the container DNS
 resolution configuration.
 <3> The percent of disk usage after which image garbage collection is always run.
 Default: 90%
 <4> The percent of disk usage before which image garbage collection is never run.
 Lowest disk usage to garbage collect to. Default: 80%
-
++
```
[the remark this answers is not in the document]
Comment (mburke5678) [reply admin_guide/manage_nodes.adoc]: @kalexand-rh No, I don't think so... Legacy text. It is just help text if you want to see more options.
https://docs.openshift.com/container-platform/3.9/admin_guide/manage_nodes.html#configuring-node-resources

File: day_two_guide/topics/pruning_images_and_containers.adoc [hunk 12]
```diff
@@ -122,8 +122,8 @@ Where `node.fs` is the file system hosting the
 triggers garbage collection processes to clean up the unused ones.
 
 There are some default values that can be tweaked if required in every node's
-configuration file located in `/etc/origin/node/node-config.yaml` within the
```
[the remark this answers is not in the document]
Comment (mburke5678) [reply day_two_guide/topics/pruning_images_and_containers.adoc]: @kalexand-rh Such a change is beyond the scope of this PR (I love saying that!). Should we create a new issue/BZ?

File: dev_guide/device_manager.adoc [hunk 13]
```diff
@@ -23,7 +23,12 @@ hardware without any upstream code changes.
 . Enable Device Manager support on the target node or nodes:
 +
 ----
-# cat /etc/origin/node/node-config.yaml
+# oc describe configmaps <name>
```
[the remark this answers is not in the document]
Comment (mburke5678) [reply dev_guide/device_manager.adoc]: @kalexand-rh Not sure. Just used what was there....

File: scaling_performance/managing_hugepages.adoc [hunk 14]
```diff
@@ -48,7 +48,13 @@ huge pages. This topic describes how.
 . `HugePages` must be set to `true` across the system:
 +
 ----
-# cat /etc/origin/node/node-config.yaml
+# oc edit configmap <name>
```
[the remark this answers is not in the document]
Comment (mburke5678) [reply scaling_performance/managing_hugepages.adoc]: See above

[the remark this answers is not in the document]
Comment (mburke5678) [reply scaling_performance/managing_hugepages.adoc]: @kalexand-rh I am going to leave this one as the following steps use `#` also. We could open an issue if we want to look into it.

**Comment by mburke5678** [comment]
@kalexand-rh Thanks for all the nits. I think we cleaned up a few years of nits that have been in the docs goodness knows how long. (I was focussing on the proper switch between node config and config map and didn't notice these nitty bits).

**Pull request merged**